rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -2199,8 +2201,24 @@ public ResponseEntity<HalResource> getLicenseObligations(
         }
 
         obligationStatusMap = projectService.setLicenseInfoWithObligations(obligationStatusMap, releaseIdToAcceptedCLI, releases, sw360User);
+        for (Map.Entry<String, ObligationStatusInfo> entry : obligationStatusMap.entrySet()) {
+            ObligationStatusInfo statusInfo = entry.getValue();
+            Set<Release> limitedSet = new HashSet<>();
+
+            for (Map.Entry<String, String> idEntry : statusInfo.getReleaseIdToAcceptedCLI().entrySet()) {
+                String relId = idEntry.getKey();
+                Release rel = releaseService.getReleaseForUserById(relId, sw360User);
```

Comment:
I think you should get the release list by creating a new function **getReleasesForUsersByIds** instead of looping on each key of the map
You can refer to **getReleasesByIdsForExport** in **ComponentHandler** to do that.

Reply:
Hi @hoangnt2 , map of obligationStatusMap already has release info but we need only limited info for releases, therefore I'm getting the releaseIds from releaseIdToAcceptedCli field which is specific to each obligation data and we cannot group or list all releaseIds at one place.

Reply:
Yes, I understood the situation. But I think we can rewrite this code as below

```
for (Map.Entry<String, ObligationStatusInfo> entry : obligationStatusMap.entrySet()) {
    ObligationStatusInfo statusInfo = entry.getValue();
    Set<Release> limitedSet = releaseService.getReleasesForUserByIds(statusInfo.getReleaseIdToAcceptedCLI().keySet(), sw360User);
    statusInfo.setReleases(limitedSet);
}
```
It will avoid querying the database too much

Reply:
Made the changes, let me know if this is fine.

Reply:
@rudra-superrr, move it to new function is not enough. You can refer to this function to get list release by their ids. https://github.com/eclipse-sw360/sw360/blob/c619d8e3ccb2a0ba1e4f530906dbd1cd96f67f27/backend/src/src-components/src/main/java/org/eclipse/sw360/components/ComponentHandler.java#L267C26-L267C51

Reply:
Done, thanks for helping.